src/Common/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -44,7 +44,7 @@ internal static SafeFileHandle Open(string path, Interop.libc.OpenFlags flags, i
                 // Open the file.
                 int fd;
                 while (Interop.CheckIo(fd = Interop.libc.open(path, flags, mode), path, isDirectory: enoentDueToDirectory,
-                    errorRewriter: errno => (errno == Interop.Errors.EISDIR) ? Interop.Errors.EACCES : errno)) ;
+                    errorRewriter: e => (e.Error == Interop.Error.EISDIR) ? new Interop.ErrorInfo(Interop.Error.EACCES) : e)) ;
```

Comment:
Per the discussion earlier about the efficiency of conversion between Error and Errno, this is another example where we end up constructing an ErrorInfo from an Error value but where (I think) we don't ever need the actual errno value, so it'd be nice not to have to pay the overheads of a managed->native call.  If you didn't want to cache the conversion table, another option to consider would be to make the conversion only happen when the alternate value is needed (changing ErrorInfo's fields to be properties).